hazelcast/src/test/java/com/hazelcast/internal/serialization/impl/compact/CompactNullablePrimitiveInteroperabilityTest.java

```diff
@@ -229,9 +277,28 @@ public void testWriteNullReadPrimitiveThrowsException() {
         assertReadNullAsPrimitiveThrowsException(serializedRecord);
     }
 
+    @Test
+    public void testWriteNullReadPrimitiveThrowsExceptionCustomSerializer() {
+        SerializationService serializationService = createSerializationService(ASerializer::new);
+
+        Data data = serializationService.toData(new A(null, new Integer[]{1, 2}));
+
+        assertThatThrownBy(() -> serializationService.toObject(data))
+                .isInstanceOf(HazelcastSerializationException.class)
+                .hasMessageContaining("Use readNullable");
+
+        SerializationService serializationService2 = createSerializationService(ASerializer::new);
```

Comment:
Why do we need a new serialization service with the same serializer?

Reply:
test delete anyway